src/server/routes/signIn.ts

```diff
@@ -247,6 +248,17 @@ const oktaSignInController = async (
     // we're authenticated track this metric
     trackMetric('OktaSignIn::Success');
 
+    // If the user was able to log in with Okta BUT NOT VALIDATED, we will still try and sign them in with Identity
+    // This is because we will initally block unvalidated users from getting Okta sessions, but still allow them
+    // Identity sessions.
+
+    if (response._embedded?.user.id) {
```

Comment:
is it possible for this to be falsy and for them not to be validated? I'm guessing if this is the case, they won't be getting signed in either way

Reply:
maybe we could cover this scenario with a test case

Reply:
yeah, as per the docs, https://developer.okta.com/docs/reference/api/authn/#request-parameters-for-primary-authentication the api endpoint for authentication can return different `_embedded` objects depending on the users current state https://developer.okta.com/docs/reference/api/authn/#transaction-state for example if a user required MFA you would get a user state returned, but also be required to take further steps to sign the user in successfully. In this case, we should also do a check for `status: 'SUCCESS'`

Reply:
I've added a new test and check for this

Reply:
thanks for the explanation — sounds good!